libraryBackend/libraryApp/api.py

```diff
@@ -160,206 +161,315 @@ def post(self, request, *args, **kwargs):
+                    "fine": i.fine,
+                    "paid": i.paid,
                 }
                 lst_issue.append(data)
-        lst_issue=lst_issue[::-1]
-        data1={
-        "id":copy.id,
-        "name":copy.book.name,
-        "condition":copy.condition,
-        "is_available":copy.is_available,
-        "issues":lst_issue
+        lst_issue = lst_issue[::-1]
+        data1 = {
+            "id": copy.id,
+            "name": copy.book.name,
+            "condition": copy.condition,
+            "is_available": copy.is_available,
+            "issues": lst_issue,
         }
         lst.append(data1)
         return Response(lst)
-    def put(self,request,id,book):
+
+    def put(self, request, id, book):
         try:
-            copy=Copy.objects.get(id=book)
+            copy = Copy.objects.get(id=book)
         except:
-            return Response("Book not found",status=HTTP_400_BAD_REQUEST)
-        condition=request.data.get("condition")
-        copy.condition=condition
+            return Response("Book not found", status=HTTP_400_BAD_REQUEST)
+        condition = request.data.get("condition")
+        copy.condition = condition
         copy.save()
-        data={
-        "id":copy.id,
-        'barcode':copy.barcode,
-        "name":copy.book.name,
-        "condition":copy.condition
+        data = {
+            "id": copy.id,
+            "barcode": copy.barcode,
+            "name": copy.book.name,
+            "condition": copy.condition,
         }
-        lst=[]
+        lst = []
         lst.append(data)
-        return Response(lst,status=HTTP_200_OK)
-    def delete(self,request,id,book):
+        return Response(lst, status=HTTP_200_OK)
+
+    def delete(self, request, id, book):
         try:
-            copy=Copy.objects.get(id=book)
+            copy = Copy.objects.get(id=book)
         except:
-            return Response("Book not found",status=HTTP_400_BAD_REQUEST)
+            return Response("Book not found", status=HTTP_400_BAD_REQUEST)
         copy.delete()
-        return Response("deleted successfully",status=HTTP_200_OK)
+        return Response("deleted successfully", status=HTTP_200_OK)
+
+
+class BookRequestView(generics.GenericAPIView):
+    serializer_class = WaitingListSerializer
+    permission_classes = [IsAuthenticated, IsTeacherStudent]
+
+    def get(self, request, book_id):
+        if Book.objects.filter(id=book_id):
+            book = Book.objects.get(id=book_id)
+            if book.available_quantity > 0:
+                copy = Copy.objects.filter(is_available=True, book=book)
+                copy = copy[0]
+                if request.user.is_student:
+                    student = Student.objects.get(email=request.user)
+                    WaitingList.objects.create(
+                        book=book,
+                        student=student,
+                        is_alerted=True,
+                        alerted_on=datetime.date.today(),
```

Comment:
This line gives the following error `RuntimeWarning: DateTimeField Image.added_on received a naive datetime (2021-04-13 14:48:22) while time zone support is active.`